common/src/main/java/com/thoughtworks/go/domain/builder/FetchPluggableArtifactBuilder.java

```diff
@@ -63,10 +66,10 @@ public FetchPluggableArtifactBuilder(RunIfConfigs conditions, Builder cancelBuil
         this.metadataFileLocationOnServer = source;
     }
 
-    public void build(DefaultGoPublisher publisher, EnvironmentVariableContext environmentVariableContext, TaskExtension taskExtension, ArtifactExtension artifactExtension) {
+    public void build(DefaultGoPublisher publisher, EnvironmentVariableContext environmentVariableContext, TaskExtension taskExtension, ArtifactExtension artifactExtension, PluginRequestProcessorRegistry pluginRequestProcessorRegistry) {
         downloadMetadataFile(publisher);
-
         try {
+            pluginRequestProcessorRegistry.registerProcessorFor(CONSOLE_LOG.requestName(), new ArtifactRequestProcessor(publisher));
```

Comment:
extract a variable for CONSOLE_LOG.requestName() ?

Reply:
I think this is fine.